java/core/src/java/org/apache/orc/impl/SchemaEvolution.java

```diff
@@ -87,7 +87,8 @@ public SchemaEvolution(TypeDescription fileSchema,
     this.fileSchema = fileSchema;
     this.isAcid = checkAcidSchema(fileSchema);
     this.includeAcidColumns = options.getIncludeAcidColumns();
-    this.readerColumnOffset = isAcid ? acidEventFieldNames.size() : 0;
+    this.readerColumnOffset = (isAcid && readerSchema != null && checkAcidSchema(readerSchema)) || !isAcid
```

Comment:
I'd suggest keeping the checkAcidSchema(readerSchema) rather than calling it multiple times. This expression needs to be simplified.